### PR TITLE
Fix/webhooks return codes

### PR DESCRIPTION
**Pull request by moraleida**
### 🎫 Ticket

[ET-1415] 

### ✔️ Checklist
- [ ] I've included a changelog entry in the readme.txt file. 
- [ ] My code is tested. 
- [ ] My code has [proper inline documentation](https://the-events-calendar.github.io/products-engineering/docs/code-standards/).

[ET-1415]: https://theeventscalendar.atlassian.net/browse/ET-1415?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Tickets/Commerce/Gateways/Stripe/Payment_Intent.php [hunk 1]
```diff
@@ -24,6 +24,8 @@ class Payment_Intent {
 	 */
 	const TEST_VALUE = 1.28;
 
+	public static $test_metadata_key = 'payment_intent_validation_test';
```
Comment (bordoni) [inline src/Tickets/Commerce/Gateways/Stripe/Payment_Intent.php:27]: Docblock.

**Review by bordoni — APPROVED** [review]

File: src/Tickets/Commerce/Gateways/Stripe/Webhooks/Handler.php [hunk 2]
```diff
@@ -52,7 +54,17 @@ public static function process_webhook_response( WP_REST_Request $request, WP_RE
 				'tickets-commerce-gateway-stripe'
 			);
 
-			return new WP_Error( 'tec-tickets-commerce-stripe-webhook-invalid-type', null, [ 'event' => $event ] );
+			$response->set_status( 200 );
+			$response->set_data(
+				sprintf(
+				// Translators: %1$s is the event id and %2$s is the event type name
```
Comment (codingmusician) [inline src/Tickets/Commerce/Gateways/Stripe/Webhooks/Handler.php:60]:
```suggestion
					// Translators: %1$s is the event id and %2$s is the event type name.
```

File: src/Tickets/Commerce/Gateways/Stripe/Webhooks/Handler.php [hunk 3]
```diff
@@ -91,14 +95,23 @@ public static function process_webhook_response( WP_REST_Request $request, WP_RE
...
 	public static function get_handler_method_for_event( $type ) {
 		$handlers = Events::get_event_handlers();
 
-		return isset( $handlers[ $type ] ) ?
-			$handlers[ $type ] :
-			new WP_Error( 200, sprintf( __( 'Webhook event was retrieved properly but %s is not a handled event.', 'event-tickets' ), esc_html( $type ) ) );
+		if ( ! isset( $handlers[ $type ] ) ) {
+			return new WP_REST_Response(
+				sprintf(
+					// Translators: %1$s is the event type name
```
Comment (codingmusician) [inline src/Tickets/Commerce/Gateways/Stripe/Webhooks/Handler.php:106]:
```suggestion
					// Translators: %1$s is the event type name.
```

File: src/Tickets/Commerce/Gateways/Stripe/Webhooks/Payment_Intent_Webhook.php [hunk 4]
```diff
@@ -20,32 +21,53 @@ class Payment_Intent_Webhook implements Webhook_Event_Interface {
...
+				$response->set_status( 200 );
+				$response->set_data(
+					__( 'Payment Intent Test Successful', 'event-tickets' )
+				);
+
+				return $response;
+			}
+
+			return new \WP_Error( 400, sprintf(
+			// Translators: %s is the payment intent id
```
Comment (codingmusician) [inline src/Tickets/Commerce/Gateways/Stripe/Webhooks/Payment_Intent_Webhook.php:56]:
```suggestion
				// Translators: %s is the payment intent id.
```

**Review by codingmusician — APPROVED** [review]

**Pull request merged**